idea-plugin/src/com/reason/ide/annotations/ErrorAnnotator.java

```diff
@@ -1,113 +1,229 @@
-      if (0 < startOffset && 0 < endOffset && startOffset < endOffset) {
-        if (LOG.isDebugEnabled()) {
-          LOG.debug("annotate " + startOffset + ":" + endOffset + " '" + message + "'");
-        }
-        TextRangeInterval range = new TextRangeInterval(startOffset - 1, endOffset - 1);
-        result.add(new BsbErrorAnnotation(info.isError, message, range, start));
-      } else {
-        if (LOG.isDebugEnabled()) {
-          LOG.debug(
-              "Failed to locate info: "
-                  + start
-                  + "->"
-                  + end
-                  + ", offsets "
-                  + startOffset
-                  + "->"
-                  + endOffset
-                  + ", info "
-                  + info);
-        }
+    // create temporary compilation directory, once
+    if (m_compilationDirectory == null) {
+      m_compilationDirectory = createTempCompilationDirectory(project);
+    }
+
+    Optional<VirtualFile> contentRoot = BsPlatform.findContentRootForFile(project, sourceFile);
+    if (!contentRoot.isPresent()) {
+      LOG.info("Unable to find BuckleScript content root.");
+      return null;
+    }
+
+    VirtualFile bsExecutableDirectory = contentRoot.get().findFileByRelativePath("lib/bs");
+    if (bsExecutableDirectory == null) {
+      LOG.info("Unable to find lib/bs for BuckleScript project.");
+      return null;
+    }
+
+    // Read bsConfig to get the jsx value and ppx
+    // @TODO register a file listener for bsconfig.json
+    VirtualFile bsConfigFile = contentRoot.get().findFileByRelativePath("bsconfig.json");
+    if (bsConfigFile == null) {
+      LOG.info("No bsconfig.json found for content root: " + contentRoot);
+      return null;
+    }
+    BsConfig config = BsConfigReader.read(bsConfigFile);
+    String jsxVersion = config.getJsxVersion();
+    String namespace = config.getNamespace();
+
+    // If a directory is marked as dev-only, it won't be built and exposed to other "dev"
+    // directories in the same project
+    // https://bucklescript.github.io/docs/en/build-configuration#sources
+    // @TODO register a file listener for build.ninja
+    BsCompiler bucklescript = ServiceManager.getService(project, BsCompiler.class);
+    Ninja ninja = bucklescript.readNinjaBuild(contentRoot.get());
+    for (String devSource : config.getDevSources()) {
+      VirtualFile devFile = contentRoot.get().findFileByRelativePath(devSource);
+      if (devFile != null && FileUtil.isAncestor(devFile.getPath(), sourceFile.getPath(), true)) {
+        ninja.addInclude(devSource);
       }
     }
 
-    return result;
+    if (!BsPlatform.findBscExecutable(project, sourceFile).isPresent()) {
+      LOG.info("Unable to find bsc.exe");
+      return null;
+    }
+
+    // Creates a temporary file on disk with a copy of the current document.
+    // It'll be used by bsc for a temporary compilation
+    Path tempFilePath = Paths.get(m_compilationDirectory.getPath(), sourceFile.getName());
```

Comment:
This file MUST be deleted at the end of the process

Reply:
instead of deleting this file at the end of the process, i was instead deleting the entire temporary directory before compiling everything.

Reply:
Not sure it works. is it ok if dir is not empty ? also createTemp might not be predictive (adding random numbers to always succeed) ? if something went wrong, you compile a dir with old files, that may produce unexpected result.... but maybe my comment is stupid

Reply:
createTemp adds a number to the prefix only if a directory already exists with the same name. By deleting the directory before we create it, we guarantee that createTemp won't add a number prefix.

Also, the last `true` parameter in `FileUtil.createTempDirectory(directoryName, null, true);` is `deleteOnExit` which will delete the temporary directory when IntelliJ exists so long as it doesn't crash